### PR TITLE
Fix too many assertions issue

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**
I just found this issue in Sonar java. And have a proposal how to make it more readable.
What do you think about this solution? Doest it look more readable for you?

### CODE REVIEW

**Comment by quentin-jaquier-sonarsource** [comment]
My main concern is that you will still execute the same number of assertions, they are just hidden behind method calls. This will remove the issue, but in the end, I feel that the problem originally reported is still present (test can fail for too many reasons).

File: external-reports/src/test/java/org/sonar/java/externalreport/PmdSensorTest.java [hunk 1]
```diff
@@ -218,3 +223,66 @@ private static String language(Path file) {
     return path.substring(path.lastIndexOf('.') + 1);
   }
 }
+
+class ExternalIssueAsserter {
```
Comment (m-g-sonar) [inline external-reports/src/test/java/org/sonar/java/externalreport/PmdSensorTest.java:227]: Just a small comment about this: I'm not a big fan of having two classes at the same level in a compilation unit, just one being not public. I would prefer to make it a public static method inside the other one, or a dedicated class in its own compilation unit.

Also, what about making it inherit from AssertJ's `AbstractAssert`?
https://joel-costigliola.github.io/assertj/assertj-core-custom-assertions.html

Comment (margarita-nedzelska-sonarsource) [reply external-reports/src/test/java/org/sonar/java/externalreport/PmdSensorTest.java]: I thought about AssertJ approach. Unfortunately, I am not sure we'll benefit much from it. Cause we still have to implement all needed assertions logic.

I also changed the logic of assertions, so there is only one point to get an assertion error and we'll have all the error messages collected.

Comment (m-g-sonar) [reply external-reports/src/test/java/org/sonar/java/externalreport/PmdSensorTest.java]: Fine by me :)

I just believe that the more we stay closer from how AssertJ operate, the higher is the chance to have it reused somewhere in our own tests.

**Pull request merged**